### PR TITLE
Add Cross-Chain IP  Registration Functionality 

### PR DESCRIPTION
**Pull request by kingster-will**
This PR introduces a new function to the `IPAssetRegistry` contract that allows for the registration of IP NFTs from different chain. 

The new function, `registerCrossChain`, takes in three parameters: `chainId`, `tokenContract`, and `tokenId`. The `chainId` represents the identifier of the blockchain network where the IP NFT resides. 

This function first checks if the `chainId` is the same as the current network's `chainId`. If they are the same, it throws an error as this function is specifically designed for cross-chain registrations.

### CODE REVIEW

File: contracts/registries/IPAssetRegistry.sol [hunk 1]
```diff
@@ -79,6 +79,34 @@ contract IPAssetRegistry is IIPAssetRegistry, IPAccountRegistry, Governable {
         emit IPRegistered(id, block.chainid, tokenContract, tokenId, name, uri, registrationDate);
     }
 
+    function registerCrossChain(uint256 chainId, address tokenContract, uint256 tokenId) external returns (address id) {
```
Comment (jdubpark) [inline contracts/registries/IPAssetRegistry.sol:82]: Can we have an internal function for `register` and `registerCrossChain`? Logic overlaps.

**Comment by jdubpark** [comment]
Might require upgradability as cross-chain requirements will evolve and might impact IPAssetRegistry.
cc @Ramarti @LeoHChen

**Comment by LeoHChen** [comment]
> Might require upgradability as cross-chain requirements will evolve and might impact IPAssetRegistry. cc @Ramarti @LeoHChen

valid concern, that's why I don't feel we shall set the IPAssetRegistry non-upgradable right now.

File: contracts/registries/IPAssetRegistry.sol [hunk 2]
```diff
@@ -79,6 +79,34 @@ contract IPAssetRegistry is IIPAssetRegistry, IPAccountRegistry, Governable {
...
+        if (block.chainid == chainId) {
```
Comment (LeoHChen) [inline contracts/registries/IPAssetRegistry.sol:83]: why revert here?  shouldn't it just work as intended?

**Comment by jdubpark** [comment]
Per some discussions, we want
- Cross-chain Registration enabled on Story Net
- Local Registration enabled on both Story Net and other chains

**Comment by LeoHChen** [comment]
> Per some discussions, we want
> 
> * Cross-chain Registration enabled on Story Net
> * Local Registration enabled on both Story Net and other chains

we need to make a decision on how to support it

**Comment by LeoHChen** [comment]
The decision is to support cross chain mint in register with chain id added for now.

File: contracts/registries/IPAssetRegistry.sol (hunk 1 above)
Comment (kingster-will) [reply contracts/registries/IPAssetRegistry.sol]: merged two functions.

File: contracts/registries/IPAssetRegistry.sol (hunk 2 above)
Comment (kingster-will) [reply contracts/registries/IPAssetRegistry.sol]: Logic changed, has removed the revert.

**Review by LeoHChen — APPROVED** [review]
LGMT, please squash all commits when merge.

**Review by jdubpark — APPROVED** [review]

**Pull request merged**